src/Linters/BaseLinter.hack

```diff
@@ -23,6 +28,29 @@ abstract class BaseLinter {
   public function __construct(private File $file) {
   }
 
+  public function setConfig(?this::TConfig $config): void {
+    $this->config = $config;
+  }
+
+  protected function getConfig(): ?this::TConfig {
+    return $this->config;
+  }
+
+  final static public function typeAssertConfig(mixed $config): this::TConfig {
+    try {
+      return TypeAssert\matches<this::TConfig>($config);
+    } catch (TypeAssert\UnsupportedTypeException $e) {
+      throw new \InvalidOperationException(
+        Str\format(
+          '%s specified an unsupported config type. See previous exception:',
+          static::class,
+        ),
+        $e->getCode(),
+        $e,
+      );
+    }
+  }
```

Comment:
It would probably be better to be more strict here, instead of "anything that TypeAssert supports is valid config" (e.g. classes), it could be something like:

- config must be a shape (can be enforced statically, see above)
- valid shape field types: int, string, bool, maaaybe float
- vec or dict of valid shape type is also a valid shape type

This can't be enforced by TypeAssert but we could probably write a test for that (instead of the runtime check you have below [1]).